src/main/java/edu/cornell/kfs/kew/mail/service/impl/CuEmailContentServiceImpl.java

```diff
@@ -0,0 +1,98 @@
+package edu.cornell.kfs.kew.mail.service.impl;
+
+import java.util.Collection;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.kuali.kfs.core.api.config.property.ConfigContext;
+import org.kuali.kfs.core.api.mail.EmailContent;
+import org.kuali.kfs.core.impl.config.property.Config;
+import org.kuali.kfs.coreservice.framework.parameter.ParameterService;
+import org.kuali.kfs.kew.actionitem.ActionItem;
+import org.kuali.kfs.kew.actionitem.ActionItemActionListExtension;
+import org.kuali.kfs.kew.doctype.bo.DocumentType;
+import org.kuali.kfs.kew.mail.service.impl.EmailContentServiceImpl;
+import org.kuali.kfs.kim.api.identity.Person;
+import org.kuali.kfs.krad.util.KRADConstants;
+import org.kuali.kfs.sys.KFSConstants;
+
+import edu.cornell.kfs.kew.CuKewConstants;
+import edu.cornell.kfs.kew.CuKewParameterConstants;
+import edu.cornell.kfs.kew.mail.service.CuEmailContentService;
+
+public class CuEmailContentServiceImpl extends EmailContentServiceImpl implements CuEmailContentService {
+
+    private static final Logger LOG = LogManager.getLogger();
+
+    private ParameterService parameterService;
+
+    @Override
+    public EmailContent generateImmediateReminder(Person user, ActionItem actionItem, DocumentType documentType) {
+        updateEmailEnvironmentIfNotRunningInProduction();
+        return super.generateImmediateReminder(user, actionItem, documentType);
+    }
+
+    @Override
+    public EmailContent generateWeeklyReminder(Person user, Collection<ActionItemActionListExtension> actionItems) {
+        updateEmailEnvironmentIfNotRunningInProduction();
+        return super.generateWeeklyReminder(user, actionItems);
+    }
+
+    @Override
+    public EmailContent generateDailyReminder(Person user, Collection<ActionItemActionListExtension> actionItems) {
+        updateEmailEnvironmentIfNotRunningInProduction();
+        return super.generateDailyReminder(user, actionItems);
+    }
+
+    private void updateEmailEnvironmentIfNotRunningInProduction() {
+        if (isProductionEnvironment()) {
+            return;
+        }
+        if (isProductionWorkflowEmailModeEnabled()) {
+            setWorkflowEmailDeploymentEnvironmentFromProperty(Config.PROD_ENVIRONMENT_CODE);
```

Comment:
Does this also need to explicitly set the environment for 'kfs-prod' when running 'production mode' in a test environment?

Reply:
Without this extra setup in the email content service, all non-Production environments will add some debug-related text to the email body. I wasn't sure whether we wanted to allow for suppressing that extra email text in a non-PROD environment.

Upon further consideration, I could update this code to avoid overwriting the setting on the content service (thus forcing non-PROD environments to always display that extra text in the email), and I could also customize that added email section to indicate what environment the email was sent from. Thoughts?

Reply:
Chad, let's chat after the standup. I was hoping the system parameter would make things clearer but did not realize it would also require the extra customizations. I thought the isProduction method was going to be used to determine behavior based on prod/non-prod env.

Reply:
After further discussion with Daniela, I will remove this extra CuEmailContentServiceImpl code so that non-PROD workflow emails will always display the extra debug content in the email text. This will bring the customization more in line with how it behaves in current Cynergy code, except that we will now allow the behavior to be configured via a parameter instead of a config property.

Reply:
Thank you very much Chad!